multiaddr.go

```diff
@@ -23,11 +23,11 @@ func NewMultiaddr(s string) (Multiaddr, error) {
 // NewMultiaddrBytes initializes a Multiaddr from a byte representation.
 // It validates it as an input string.
 func NewMultiaddrBytes(b []byte) (Multiaddr, error) {
-	s, err := bytesToString(b)
-	if err != nil {
+	if err := validateBytes(b); err != nil {
```

Comment:
is `recover` expensive? or just `panic`? i'd be pretty worried of running this in production still.

Let's add a `recover` here and if it hits, log an error. if we dont see it happen for a while in production we remove it later. It would be way worse to fail hard because of a panic here than the small slowdown this involves